### ISSUE TITLE
kubectl get --export doesn't strip CRDs

### ISSUE THREAD

**Issue by errordeveloper** [kubernetes/kubernetes#50454]
It appears that `ObjectMeta` is not stripped when attempting to use `kubectl get --export` with a CRD.

Here is a quick proof:
```
> kubectl --context istio-test get IstioConfigs --output yaml | shasum
3a16cff517d0c12765fa7de3fb5dc9343105b159  -

> kubectl --context istio-test get IstioConfigs --output yaml --export | shasum
3a16cff517d0c12765fa7de3fb5dc9343105b159  -

> kubectl --context istio-test get deployments reviews-v1 --output yaml | shasum
4ae3556232657b10b52bf28ed3dbb99576b783fe  -

> kubectl --context istio-test get deployments reviews-v1 --output yaml --export | shasum
ac7943d32b8436cc7591bed26b2dd3b22a72b6a8  -
```

If you examine the output for the CRD, you will see that `creationTimestamp`, `uid`, and `resourceVersion` are untouched, which is not the case for non-CRDs, e.g. deployments.

**Comment by xingzhou** [comment]
I'd like to take a look at this issue

**Comment by xingzhou** [comment]
Had some investigation today, seems now the option `export` is only applied for single resource `get`, @errordeveloper, not sure if you can try the following command:
```
kubectl --context istio-test get IstioConfigs <the_name_of_a_single_istioConfig> --export --output yaml
```
for the above one, as it's using generic store object, it should strip the field like `creationTimeStamp`

**Comment by errordeveloper** [comment]
`/lifecycle frozen`

**Comment by nikhita** [comment]
@errordeveloper Like @xingzhou mentioned, we can only use `--export=true` against a singular resource name. There is already an issue open for the same: https://github.com/kubernetes/kubernetes/issues/24855#issuecomment-329889844.

Did you face any other issues with respect to custom objects?

**Comment by errordeveloper** [comment]
I haven't had a chance to try this again yet, but if you are quite sure
this should work fine, please feel free to close.

**Comment by liggitt** [comment]
/close
in favor of https://github.com/kubernetes/kubernetes/issues/24855

**Comment by errordeveloper** [comment]
I can confirm that exporting a single resource works fine, e.g. with:
```
kubectl get seldondeployments.machinelearning.seldon.io --output yaml --export seldon-deployment-example
```

**Comment by errordeveloper** [comment]
@nikhita that'd be nice! And actually, after I exported an object and noticed that the issue I reported here originally had been fixed, I had to still [manually strip](https://github.com/errordeveloper/seldon-gitops/commit/9bc9c13f08afc8ee0319ea86cb888ed39c57d930) some fields (a lot of which were empty strings). I understand that `--export` doesn't in its definition may not imply human-friendly, but it'd be nice if we gave this some real thought!
